WordPress/Sniffs/NamingConventions/ValidPostTypeSlugSniff.php

```diff
@@ -0,0 +1,208 @@
+		return parent::register();
+	}
+
+	/**
+	 * Process the parameter of a matched function.
+	 *
+	 * Errors on invalid post type names when reserved keywords are used,
+	 * the post type is too long, or contains invalid characters.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		// Check for non string based slug parameter (we cannot determine if this is valid).
+		$contains_text_string_token = $this->phpcsFile->findNext( $this->valid_tokens, $parameters[1]['start'], $parameters[1]['end'] + 1 );
+		if ( ! $contains_text_string_token || ( $contains_text_string_token && $this->phpcsFile->findNext( Tokens::$functionNameTokens, $parameters[1]['start'], $parameters[1]['end'] + 1 ) !== false ) ) {
+			$this->phpcsFile->addWarning(
+				'Dynamic post type slug found. It is not possible to determine the validity of this slug. Found: %s.',
+				$stackPtr,
+				'Dynamic',
+				array(
+					$parameters[1]['raw'],
+				),
+				3
+			);
+			return;
+		}
+
+		// Handling string interpolation.
+		$stringPos = $this->phpcsFile->findNext( Tokens::$textStringTokens, $parameters[1]['start'], $parameters[1]['end'] + 1, false, null, true );
+		if ( false === $stringPos ) {
+			return;
+		}
+
+		$post_type = $this->strip_quotes( $this->tokens[ $stringPos ]['content'] );
+		$data      = array(
+			$this->tokens[ $stringPos ]['content'],
+		);
+
+		// Warn for dynamic parts in the slug parameter.
+		if ( 'T_DOUBLE_QUOTED_STRING' === $this->tokens[ $stringPos ]['type'] || ( 'T_HEREDOC' === $this->tokens[ $stringPos ]['type'] && strpos( $this->tokens[ $stringPos ]['content'], '$' ) !== false ) ) {
+			$this->phpcsFile->addWarning(
+				'register_post_type() slug may or may not get too long with dynamic contents and could contain invalid characters. Found: %s.',
+				$stringPos,
+				'PartiallyDynamic',
+				$data
+			);
+			$post_type = $this->strip_interpolated_variables( $post_type );
+		}
+
+		// Error for using reserved slug names.
+		if ( strlen( $post_type ) === 0 ) {
+			$this->phpcsFile->addError(
+				'register_post_type() called without a post type slug. The slug must be a non-empty string.',
+				$stringPos,
+				'Empty'
+			);
+		}
+
+		if ( isset( $this->reserved_names[ $post_type ] ) ) {
+			$this->phpcsFile->addError(
+				'register_post_type() called with reserved post type %s. Reserved post types should not be used as they interfere with the functioning of WordPress itself.',
+				$stringPos,
+				'Reserved',
+				$data
+			);
+		}
+
+		if ( preg_match( self::POST_TYPE_WP_PREFIX, $post_type ) !== 0 ) {
```

Comment:
This should probably be an `elseif`, otherwise `wp_block` would give two errors instead of just the one.

```php
register_post_type( 'wp_block', array() );
```

Reply:
Please add the above test case to the unit test file so we can safeguard that this only throws one error, not two.

Reply:
Additional test has been added.